include/deal.II/fe/fe_values.h

```diff
@@ -3508,6 +3508,86 @@ class FESubfaceValues : public FEFaceValuesBase<dim,spacedim>
 #ifndef DOXYGEN
 
 
+// Convenience template functions used in fe_values.cc and its instantiation
+// files
```

Comment:
As far as I see, we only use the functions below in `source/fe/fe_values.cc`. Can you elaborate on why you want to move them here?

Reply:
These function templates are not used in every translation unit with this PR, so I moved them to avoid 'unused function' warnings.

Reply:
It should not be the function templates but only  the function
```
IndexSet::value_type
get_vector_element (const IndexSet &is, const types::global_dof_index cell_number)
```
that is responsible for an `unused function` warning as far as I (verified by `clang` and `gcc`).
If we just mark this function (and for consistency also the template version) as `inline`, the warnings should vanish and we don't have to pollute the header file. Would you be willing to do that instead?

Reply:
Good point: I didn't realize that marking these as inline would fix it.

I'll post an alternative patch in a moment.

Reply:
OK, I marked all of them as `inline`. That fixes the warnings.

Reply:
Thanks!